src/misc/msccl/msccl_status.cc

```diff
@@ -27,18 +27,20 @@ static mutex rankStatesMutex;
 static unordered_map<int, shared_ptr<mscclRankState>> rankStates;
 
 static inline mscclRankState& mscclGetRankState(int rank) {
-  static thread_local shared_ptr<mscclRankState> threadRankState = make_shared<mscclRankState>();
-
+  // In the unlikely case of negative rank, return a per-thread state
   if (rank < 0) {
+    static thread_local shared_ptr<mscclRankState> threadRankState(new mscclRankState());
```

Comment:
I don't think the negative `rank` is an unlikely case, although this might still be fine.

(`rank` has a default value of `-1` and early in the code this function will be called without a rank.)